paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -347,7 +345,7 @@ internal abstract class BaseSheetViewModel(
 
         if (stripeIntent is PaymentIntent) {
             runCatching {
-                savedStateHandle[SAVE_AMOUNT] = Amount(
+                _amount.value = Amount(
                     requireNotNull(stripeIntent.amount),
                     requireNotNull(stripeIntent.currency)
                 )
```

Comment:
Can we make `amount: StateFlow<Amount?>` just a value derived from `stripeIntent` via a `map { … }`?